docs/how-tos/typescript_consume.md

```diff
@@ -0,0 +1,161 @@
+            body: JSON.stringify({
+                query: request.text,
+                variables,
+            }),
+        });
+
+        return await response.json();
+    };
+
+    function createRelayEnvironment() {
+        return new Environment({
+        network: Network.create(fetchFn),
+        store: new Store(new RecordSource()),
+        });
+    }
+
+    export const RelayEnvironment = createRelayEnvironment();
+
+    ```
+
+!!! info
+
+    Refer to the [Relay Environment Docs](https://relay.dev/docs/v1.6.2/relay-environment/) to learn more.
+
+## Generate TypeScript Types
+
+Relay generated TypeScript types automatically for your queries, based on the GraphQL schema.
+For examples, if you are querying a Person type form the GraphQL schema, Relay will generate types for you.
+
+!!! example "GraphQL Query"
+
+    ```graphql
+    query PersonQuery {
+        person {
+            id
+            firstName
+            lastName
+        }
+    }
+    ```
+
+Relay will automatically generate a TypeScript type for this Query when you run the Relay compiler.
+
+!!! example "Generated Type"
+
+    ```typescript
+    // __generated__/PersonQuery.graphql.ts
+
+    export interface PersonQuery {
+        person: {
+            id: string;
+            firstName: string;
+            lastName: string;
+        } | null;
+    }
+    ```
+
+## Fetch Data Using Relay
+
+In Relay, you defing your GraphQL queries within components using the `useLazyLoadQuery` or `useFragment` hooks. Relay then fetched and managed data for you.
+
+To fetch data, we define a query in our component and use Relay's `useLazyLoadQuery` hook to execute it. The query is defined using the GraphQL tag provided by `babel-plugin-relay`.
+
+!!! example "Fetch Data with Relay"
+
+    ```ts
+    import { graphql, useLazyLoadQuery } from "react-relay";
+    import { PersonQuery } from './__generated__/PersonQuery.graphql'; // generated by relay compiler
+
+    const PersonComponent = () => {
+        const data = useLazyLoadQuery<PersonQuery>(
+            graphql`
+            query PersonQuery {
+                person {
+                id
+                firstName
+                lastName
+                }
+            }
+            `,
+            {}
+        );
+        if (!data.person) {
+            return <div>Loading...</div>;
+        }
+        return (
+            <div>
+                <h1>{data.person.firstName} {data.person.lastName}</h1>
+            </div>
+        );
+    };
+
+    export default PersonComponent;
+    ```
+
+## Mapping the Response
+
+With Relay, you don't need to manually map the response to TypeSciprt models as the Relay handles this internally and generated types for each query.
+Relay will ensure the data conforms to the types defined in your GraphQL schema.
+It will map the response to the PersonQuery type and ensure that your TypeScript code only access the field that match the GraphQL schema, such as id, firstName and lastName.
```

Comment:
```suggestion
It will map the response to the PersonQuery type and ensure that your TypeScript code only access the fields that match the GraphQL schema, such as id, firstName and lastName.
```